cinn/hlir/framework/graph_compiler.cc

```diff
@@ -1420,23 +1420,32 @@ std::vector<ir::LoweredFunc> GetFuncFromImpl(const std::shared_ptr<OpImpl>& impl
   // 4. Optimize the LoweredFunc
   VLOG(3) << "expr_pack.size() is : " << expr_pack.size();
   std::vector<ir::LoweredFunc> res;
-  for (int i = 0; i < expr_pack.size(); i++) {
-    if (funcs.size() > expr_pack.size()) {
-      auto new_args  = lang::GetArgs(funcs[i]->body, input_output_nodes);
-      funcs[i]->args = new_args;
+  CHECK_GE(funcs.size(), expr_pack.size());
+  if (funcs.size() > expr_pack.size()) {
+    for (int i = 0; i < funcs.size(); i++) {
+      for (int j = 0; j < expr_pack.size(); j++) {
+        Expr temp = expr_pack[j];
+        if (temp == funcs[i]->body) {
+          auto new_args  = lang::GetArgs(funcs[i]->body, input_output_nodes);
+          funcs[i]->args = new_args;
+          res.push_back(funcs[i]);
+        }
+      }
     }
-    auto temp_buffers   = lang::GetTempBuffers(all_arg_tensors, stages, funcs[i]->body);
-    funcs[i]->temp_bufs = temp_buffers;
-    funcs[i]->PrepareBufferCastExprs();
-    res.push_back(funcs[i]);
+  } else {
+    res = funcs;
   }
+  CHECK_EQ(res.size(), expr_pack.size());
+
   for (int i = 0; i < res.size(); i++) {
 #ifdef CINN_WITH_CUDA
     optim::OptimizeExprGPU(&(res[i]->body));
 #endif
+    auto temp_buffers = lang::GetTempBuffers(all_arg_tensors, stages, res[i]->body);
+    res[i]->temp_bufs = temp_buffers;
+    res[i]->PrepareBufferCastExprs();
```

Comment:
对res操作后又将res重新赋值，感觉有点难读啊，建议改名。比如前面的res可以改名成`lower_funcs`，最终返回值才叫res：
```suggestion
    lower_funcs[i]->PrepareBufferCastExprs();
    res.emplace_back(optim::Optimize(Expr(lower_funcs[i]), target, false).as_lowered_func_ref());
```

Reply:
Done, thanks